s3update.go

```diff
@@ -17,6 +18,8 @@ import (
 	"github.com/mitchellh/ioprogress"
 )
 
+var remoteFileSize int64
```

Comment:
I would replace this global variable by variable local to the function, and pass it as argument to the function that requires it.
It can lead to issue on concurrent calls.